grails-app/domain/org/pih/warehouse/inventory/CycleCountCandidate.groovy

```diff
@@ -0,0 +1,69 @@
+package org.pih.warehouse.inventory
+
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.core.Tag
+import org.pih.warehouse.product.Product
+
+class CycleCountCandidate {
+
+    String id
+
+    Inventory inventory
+
+    Product product
+
+    Location facility
+
+    String abcClass
+
+    Integer inventoryItemCount
+
+    String internalLocations
+
+    Integer quantityOnHand
+
+    Integer quantityAvailable
+
+    Integer negativeItemCount
+
+    Date dateLastCount
+
+    Date dateNextCount
+
+    Date dateLatestInventory
+
+    static constraints = {
+        version false
+        table "cycle_count_session"
+    }
+
+
+    Map toJson() {
```

Comment:
I'm surprised this is performant since we're fetching products, then their categories, catalogs, and tags. It seems like it would result in a lot of queries.

But if it's quick enough like you said, I'm fine with it for now.

Reply:
@EWaterman yeah, since the results are paginated, it's not that slow. The slowest part of any of those queries is the part when the view is re-built. 
As soon as the PR is approved I will prepare somewhat a description of the endpoint with avg times. Anyway, it was not more than 3s in any of the scenarios.